hail/src/main/scala/is/hail/expr/ir/MatrixIR.scala

```diff
@@ -80,6 +80,8 @@ abstract sealed class MatrixIR extends BaseIR {
 
   def rvRowPType: PStruct = typ.rvRowType.physicalType  // remove when all nodes have type rules
 
+  def physicalRowKey: IndexedSeq[String] = typ.rowKey
```

Comment:
What's physical about this?

Reply:
There are also bugs that exist because this isn't available statically, e.g. https://github.com/hail-is/hail/issues/4874

Reply:
Got it, thanks!

Reply:
Wait, but it is currently the virtual key.  You're going to override it in the suitable places?  Or make it be the physical type's key once that is in place?

Reply:
this is a placeholder. could also make it ??? I suppose, would you prefer that?

Reply:
Yeah, that might be better.  Someone might use it before it is correct.

Reply:
Sure, will fix. But this _is_ totally safe -- all it could do is shorten the key unintentionally.